src/client/datascience/interactive-window/nativeInteractiveWindow.ts

```diff
@@ -500,29 +512,106 @@ export class NativeInteractiveWindow implements IInteractiveWindowLoadable {
+
+        let result = true;
 
         // Request execution
-        await this.commandManager.executeCommand('notebook.cell.execute', {
-            ranges: [{ start: notebookDocument.cellCount - 1, end: notebookDocument.cellCount }],
-            document: notebookDocument.uri,
-            autoReveal: true
-        });
-        return true;
+        if (!debugInfo) {
+            await this.kernel?.executeCell(notebookCell);
+        } else {
+            const notebook = this.kernel?.notebook;
+            if (!notebook) {
+                return false;
+            }
+
+            try {
+                const finishedAddingCode = createDeferred<void>();
+
+                // Before we try to execute code make sure that we have an initial directory set
+                // Normally set via the workspace, but we might not have one here if loading a single loose file
+                if (file !== Identifiers.EmptyFileName) {
+                    await notebook.setLaunchingFile(file);
+                }
+
+                if (debugInfo) {
+                    // Attach our debugger based on run by line setting
+                    if (debugInfo.runByLine && debugInfo.hashFileName) {
+                        await this.jupyterDebugger.startRunByLine(notebook, debugInfo.hashFileName);
+                    } else if (!debugInfo.runByLine) {
+                        await this.jupyterDebugger.startDebugging(notebook);
+                    } else {
+                        throw Error('Missing hash file name when running by line');
+                    }
+                }
+
+                // If the file isn't unknown, set the active kernel's __file__ variable to point to that same file.
+                await this.setFileInKernel(file, notebookDocument);
+
+                const owningResource = this.owningResource;
+                // await this.kernel?.executeHidden(code, file, notebookDocument);
+                const observable = this.kernel!.notebook!.executeObservable(code, file, line, id ?? uuid(), false);
+                const temporaryExecution = this.notebookController!.controller.createNotebookCellExecution(
+                    notebookCell
+                );
+                temporaryExecution?.start();
+
+                // Sign up for cell changes
+                observable.subscribe(
+                    async (cells: ICell[]) => {
+                        // Combine the cell data with the possible input data (so we don't lose anything that might have already been in the cells)
+                        const combined = cells.map(this.combineData.bind(undefined, data));
+
+                        // Then send the combined output to the UI
+                        const converted = (combined[0].data as nbformat.ICodeCell).outputs.map(
+                            cellOutputToVSCCellOutput
+                        );
+                        await temporaryExecution.replaceOutput(converted);
```

Comment:
This code for interactive window debugging is identical to the interactiveBase.ts version, except here we update the `NotebookCellOutput` using VS Code APIs instead of posting `ICell`s to the webview.

Note, I'm making an assumption here that we'll only ever get an `ICell` array of length one back from the observable change callback. Is it possible for a cell execution to update the output of more than one cell?

Reply:
> Is it possible for a cell execution to update the output of more than one cell?

Yes. I think we should try to use the executeCell in `IKernel`.
Any reason why we cant use that instead of the executeObservable.
(after all tat's what we're doing in the non-debug case).